agent/src/agent/api/routes/pipeline.py

```diff
@@ -167,3 +167,17 @@ def pipeline_offset_changed(pipeline_id: str):
         float(request.get_json()['offset'])
     )
     return jsonify('')
+
+
+@pipelines.route('/pipelines/<pipeline_id>/watermark', methods=['GET'])
+@needs_pipeline
+def calculate_watermark(pipeline_id: str):
+    pipeline_ = pipeline.repository.get_by_id(pipeline_id)
+    if not pipeline_.offset:
+        return jsonify('')
```

Comment:
maybe we should return something more meaningful so we could at least write it to logs that there is no offset and no watermark/.

Reply:
added logs in streamsets